neo/NeoSystem.cs

```diff
@@ -37,13 +39,35 @@ public NeoSystem(Store store)
             Plugin.LoadPlugins(this);
         }
 
-        public void Dispose()
+        public void Dispose(ConfirmSafeToShutdownMethod confirmShutdownIsSafe)
         {
             RpcServer?.Dispose();
             ActorSystem.Stop(LocalNode);
+
+            Blockchain blockchain = Neo.Ledger.Blockchain.Singleton;
+            uint latestHeight = blockchain.Height;
+            uint latestHeaderHeight = blockchain.HeaderHeight;
+            bool isSafeToShutdown;
+            do
+            {
+                uint prevHeight = latestHeight;
+                uint prevHeaderHeight = latestHeaderHeight;
+                // Wait for the Blockchain to settle in case blocks are still being persisted.
+                Thread.Sleep(1000);
```

Comment:
Maybe I’ll also wait for Blockchain to be in the idle state instead of only relying on the block number having not incremented during the last second.

Reply:
Implemented this.